### ISSUE TITLE
signal 11 (SIGSEGV) error /signal Segmentation fault (11)  error when use RedisCluster

### ISSUE THREAD

**Issue by neilqin** [phpredis/phpredis#1118]
centos6.5
php 7.1.0
phpredis 3.1.0 3.1.1

when use RedisCluster to connect redis cluster to set get value , is not work

in nginx1.6 + php7.1.0-fpm:
sometimes is ok ,sometimes is wrong,see php fpm log ,
warning and notice as below:

child 5288 exited on signal 11 (SIGSEGV) after 2.8 seconds from start
child 5318 started

in apache2.2+php7.0.6:
it dose not work, see error log:
[notice] child pid 4073 exit signal Segmentation fault (11)

at last ,i use "phpredis.3.0.rc1 " in php7 ,is ok

**Comment by michael-grunder** [comment]
Hey,

We fixed a sporadic redis cluster segfault with this commit 162d882 which is not yet published to PECL.

I'm not sure if that is what's causing your crash but it's possible.

You could try using the newer unstable code or just run 3.0.0 until we push a new version.

**Comment by OwlyCode** [comment]
Hello,

I had a segfault issue with this code : 

```php
<?php

require __DIR__ . '/vendor/autoload.php';

$c = new RedisCluster(null, ['redis_cluster:7000', 'redis_cluster:7001', 'redis_cluster:7002', 'redis_cluster:7003', 'redis_cluster:7004', 'redis_cluster:7005']);
$c->multi(\Redis::PIPELINE);

$c->set("key", 'value');
$c->get("key");

$results = $c->exec();
var_dump($results);

echo "done";
```

And even sometime a `zend_mm_heap corrupted` error.

I compiled from the develop branch and now it works fine.

**Comment by michael-grunder** [comment]
There is a known issue in 3.1.1 which is fixed in develop and in the 3.1.2 branch.  I'm going to be pushing the 3.1.2 branch to PECL shortly.
